### PR TITLE
refactor(connlib): Remove unused `on_error` callback

### LINKED ISSUE #2867: remove `on_error` call
This has been brought up by @thomaseizinger before but we don't have any case were we do anything but log the error with the `on_error` callback. If it's a fatal error we use `on_disconnect` and we are already logging errors from connlib.

Until we have any non-fatal error that we want to show the users we probably should remove this callback that only adds boiler plate.

I'm also fine having the API still there but getting rid of the callbacks.

WDYT @jamilbk @roop ?

### LINKED ISSUE #3161: feat: clarify "Portal error" in logs
**Is your feature request related to a problem? Please describe.**
I talked to @jamilbk about this in Slack, he said this happens if a resource doesn't have any online gateways we can use to reach it. It's not showing me which resource is causing the error, and I'm not sure what the action is to fix it, if any.

**Describe the solution you'd like**
Add more info to the log and clarify what the solution might be

**Describe alternatives you've considered**
Haven't considered any.

**Additional context**
image

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes #3161 
Fixes #2867

### CODE REVIEW

**Review by ReactorScram — APPROVED** [review]
LGTM, thanks

File: rust/connlib/clients/shared/src/control.rs [hunk 1]
```diff
@@ -259,7 +256,10 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
 
                 tokio::spawn(async move {
                     if let Err(e) = upload(path.clone(), url).await {
-                        tracing::warn!("Failed to upload log file: {e}");
+                        let path_display = path.display();
+                        tracing::warn!(
+                            "Failed to upload log file at path {path_display}: {e}. Not retrying."
+                        );
```
Comment (conectado) [inline rust/connlib/clients/shared/src/control.rs]:
```suggestion
                        tracing::warn!(
                            "Failed to upload log file at path {path_display}: {e}. Not retrying.", path_display=path.display()
                        );
```

File: rust/connlib/clients/shared/src/control.rs [hunk 2]
```diff
@@ -278,13 +278,7 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
...
                 let Ok(resource_id) = reference.parse::<ResourceId>() else {
-                    tracing::error!(
-                        "An offline error came back with a reference to a non-valid resource id"
-                    );
-                    let _ = self
-                        .tunnel
-                        .callbacks()
-                        .on_error(&Error::ControlProtocolError);
+                    tracing::warn!("The portal responded with an Offline error. Is the Resource associated with any online Gateways? ResourceId: {reference}");
                     return Ok(());
```
Comment (conectado) [inline rust/connlib/clients/shared/src/control.rs]: the `reference` no longer points to a resource id

**Review by conectado — APPROVED** [review]
LGTM! left some non-blocking comments.

File: rust/connlib/clients/shared/src/control.rs (hunk 2 above)
Comment (jamilbk) [reply rust/connlib/clients/shared/src/control.rs]: 👍

**Pull request merged**